node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java

```diff
@@ -415,6 +426,27 @@ private static Allocation ofIpv4(String ipAddress, NameResolver resolver) {
             return new Allocation(hostname4, Optional.of(addresses.get(0)), Optional.empty());
         }
 
+        private static Allocation fromAddress(Address address, NameResolver resolver, IpAddresses.Protocol protocol) {
+            Optional<String> ipv4Address = resolveOptional(address.hostname(), resolver, RecordType.A);
+            if (protocol != IpAddresses.Protocol.ipv6 && ipv4Address.isEmpty())
```

Comment:
Avoid resolving `A` records in the first place if protocol is `IPv6`?

Reply:
:thinking: or, verify no A records are found for a supposedly IPv6-only host?  We expect later resolutions in host admin or various tools to blindly find all A and AAAA records, so let's ensure they are exactly as expected?  Implemented the latter.